app/controllers/api_controller/normalizer.rb

```diff
@@ -111,8 +111,12 @@ def normalize_encrypted
     def normalize_select_attributes(obj, opts)
       if opts[:render_attributes].present?
         opts[:render_attributes]
+      elsif obj.respond_to?(:attributes) && obj.class.respond_to?(:virtual_attribute?)
+        obj.attributes.keys.select { |k| !obj.class.virtual_attribute?(k) }
```

Comment:
Just curious, would something like obj.attributes.keys - obj.class.virtual_attributes.keys (or equivalent for second part) be faster ?